Jellyfin.Server.Implementations/Migrations/20210518000710_NewImageModel.cs

```diff
@@ -0,0 +1,269 @@
+                    schema: "jellyfin",
+                    table: "ImageInfos",
+                    type: "INTEGER",
+                    nullable: true);
+
+                int i = (int)ImageType.Profile;
+
+                migrationBuilder.Sql("UPDATE ImageInfos SET Type = " + i.ToString());
+
+                migrationBuilder.AlterColumn<int>(
+                    name: "Type",
+                    schema: "jellyfin",
+                    table: "ImageInfos",
+                    nullable: false);
+
+                migrationBuilder.RenameColumn(
+                    name: "LastModified",
+                    schema: "jellyfin",
+                    table: "ImageInfos",
+                    newName: "LastModifiedDate");
+
+                migrationBuilder.AddColumn<DateTime>(
+                    name: "AddedDate",
+                    schema: "jellyfin",
+                    table: "ImageInfos",
+                    type: "TEXT",
+                    nullable: true);
+
+                migrationBuilder.Sql("UPDATE ImageInfos SET AddedDate = (SELECT i.LastModifiedDate FROM ImageInfos i WHERE i.Id = ImageInfos.Id)");
+
+                migrationBuilder.AlterColumn<DateTime>(
+                    name: "AddedDate",
+                    schema: "jellyfin",
+                    table: "ImageInfos",
+                    nullable: false);
+
+                migrationBuilder.AddColumn<DateTime>(
+                    name: "FileCreationDate",
+                    schema: "jellyfin",
+                    table: "ImageInfos",
+                    type: "TEXT",
+                    nullable: true);
+
+                migrationBuilder.AddColumn<DateTime>(
+                    name: "FileModificationDate",
+                    schema: "jellyfin",
+                    table: "ImageInfos",
+                    type: "TEXT",
+                    nullable: true);
+
+                migrationBuilder.AddColumn<string>(
+                    name: "Blurhash",
+                    schema: "jellyfin",
+                    table: "ImageInfos",
+                    type: "TEXT",
+                    nullable: true);
+
+                migrationBuilder.DropPrimaryKey(
+                    name: "PK_ImageInfos",
+                    schema: "jellyfin",
+                    table: "ImageInfos");
+
+                migrationBuilder.AddColumn<Guid>(
+                    name: "UUID",
+                    schema: "jellyfin",
+                    table: "ImageInfos",
+                    type: "TEXT",
+                    nullable: true,
+                    defaultValue: new Guid("00000000-0000-0000-0000-000000000000"));
+
+                var triggerQuery = @"CREATE TRIGGER AutoGenerateGUID
+                                    AFTER UPDATE ON ImageInfos
+                                    FOR EACH ROW
+                                    BEGIN
+                                        UPDATE ImageInfos SET UUID = (select hex(randomblob(4)) || '-' || hex(randomblob(2))
```

Comment:
How is the performance on this exactly? Can't say I'm a fan of triggers

Reply:
@cvium Both this migration and the ``AddIndexesAndCollations`` one (the only two pending migrations since 10.7.X) run in less than 0.5 seconds in my Raspberry Pi. This trigger is just a "workaround" around the fact that we can't do ``SELECT``s in migrationBuilder and SQLite doesn't support functions, so I can't do a programatic update that allows me to generate a GUID for each row in the table.

The trigger is the best "programatic" method SQLite has for the kind of behaviour we have, and it's dropped as soon as all the migration procedure finishes, so it won't affect performance outside the migration process itself. New installations will also have no performance hit, as the tables will be empty.

Reply:
I missed the line where you drop the trigger. What makes you say we can't do SELECTs in migrationbuilder though?

Reply:
@cvium We can do SELECTs in subqueries (in fact I do them in many of the other statements as subqueries), however we can't get the returned values in C#, so we can't do a ``foreach (var row in query)`` like we do in our custom migration runner, as migrationBuilder returns an instance of [``operationBuilder``](https://docs.microsoft.com/es-es/dotnet/api/microsoft.entityframeworkcore.migrations.operations.builders.operationbuilder-1?view=efcore-5.0).

With raw SQL, I'm not able to do an UPDATE operation where the SELECT that generates UUIDs generates an unique UUID for each row. The only way of doing it is using a trigger, which runs an statement per updated row.